pkgs/io_file/test/read_as_bytes_test.dart

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@TestOn('posix')
+library;
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:isolate';
+import 'dart:typed_data';
+
+import 'package:io_file/io_file.dart';
+import 'package:io_file/posix_file_system.dart' show blockSize, maxReadSize;
+import 'package:stdlibc/stdlibc.dart' as stdlibc;
+import 'package:test/test.dart';
+
+import 'test_utils.dart';
+
+/// Writes data to a FIFO.
+///
+/// The data is written in a seperate [Isolate] to prevent deadlock during
+/// read operations.
+class FifoWriter {
+  final SendPort _sendChannel;
+
+  FifoWriter._(this._sendChannel);
+
+  static Future<FifoWriter> create(String path) async {
+    final p = ReceivePort();
+    await Isolate.spawn<SendPort>((port) {
+      final receivePort = ReceivePort();
+      port.send(receivePort.sendPort);
+
+      final fd = stdlibc.open(
+        path,
+        flags: stdlibc.O_WRONLY | stdlibc.O_CLOEXEC,
+      );
+
+      if (fd == -1) {
+        throw AssertionError('could not open fifo: ${stdlibc.errno}');
+      }
+      late final StreamSubscription<dynamic> subscription;
+      Future<void> pause(Duration d) async {
+        subscription.pause();
+        await Future<void>.delayed(d);
+        subscription.resume();
+      }
+
+      subscription = receivePort.listen(
+        (message) => switch (message) {
+          Uint8List data => stdlibc.write(fd, data),
+
+          Duration d => pause(d),
+          null => stdlibc.close(fd),
+
+          final other => throw UnsupportedError('unexpected message: $other'),
+        },
+      );
+    }, p.sendPort);
+    return FifoWriter._((await p.first) as SendPort);
+  }
+
+  void write(Uint8List data) {
+    _sendChannel.send(data);
+  }
+
+  void delay(Duration d) {
+    _sendChannel.send(d);
+  }
+
+  void close() {
+    _sendChannel.send(null);
+  }
+}
+
+void main() {
+  //TODO(brianquinlan): test with a very long path.
+
+  group('readAsBytes', () {
+    late String tmp;
+
+    setUp(() => tmp = createTemp('readAsBytes'));
+
+    tearDown(() => deleteTemp(tmp));
+
+    test('non-existant file', () {
+      expect(
+        () => fileSystem.readAsBytes('doesnotexist'),
+        throwsA(
+          isA<PathNotFoundException>()
+              .having((e) => e.message, 'message', 'open failed')
+              .having(
+                (e) => e.osError?.errorCode,
+                'errorCode',
+                2, // ENOENT
```

Comment:
Would this be more readable as `stdlibc.ENOENT` instead of using a comment (ditto below)

Reply:
This is much better!